src/schema/v2/artworksForUser/helpers.ts

```diff
@@ -33,7 +43,11 @@ export const getNewForYouRecs = async (
           }
         }
       `,
-  })()
+  }
+
+  const vortexResult = xImpersonateUserID
+    ? await vortexGraphQLUnauthenticatedLoader(query)()
+    : await vortexGraphQLAuthenticatedLoader!(query)()
```

Comment:
Rather than use `!`, why not follow the TS suggestion and do an exist-y check? If an un-authenticated request that doesn't specify an impersonate user hits this endpoint, it'll try to use this loader, and probably fail with an odd error cause this loader doesn't exist.

Whereas TS is telling you that, so you can specifically do `if (!authenticatedLoader) throw new Error('You must be logged in')` or something like that.

In general we overuse `!` a lot and it should probably only be used rarely.

This isn't a blocker in this case however, b/c while following TS will enable you to show a more sensible error message, we know the request can't succeed if there's no impersonation _and_ there's no access token.